src/main/java/mate/dao/CarDaoImpl.java

```diff
@@ -157,7 +157,7 @@ public List<Car> getAllByDriver(Long driverId) {
     private void insertAllDrivers(Car car) {
         Long carId = car.getId();
         List<Driver> drivers = car.getDrivers();
-        if (drivers.size() == 0) {
+        if (drivers == null) {
             return;
         }
```

Comment:
yes, the problem was here
and maybe it would be better to have `drivers == null || drivers.size() == 0` as a condition